tests/contrib/wsgi/test_wsgi.py

```diff
@@ -0,0 +1,201 @@
+
+    with override_config("wsgi", dict(distributed_tracing=False)):
+        app = TestApp(wsgi.DDWSGIMiddleware(application, tracer=tracer))
+        resp = app.get("/", headers={"X-Datadog-Parent-Id": "1234", "X-Datadog-Trace-Id": "4321"})
+        assert config.wsgi.distributed_tracing is False
+        assert resp.status == "200 OK"
+        assert resp.status_int == 200
+
+        spans = tracer.writer.pop()
+        assert len(spans) == 4
+        root = spans[0]
+        assert root.name == "wsgi.request"
+        assert root.trace_id != 4321
+        assert root.parent_id != 1234
+
+
+def test_query_string_tracing(tracer):
+    with override_http_config("wsgi", dict(trace_query_string=True)):
+        app = TestApp(wsgi.DDWSGIMiddleware(application, tracer=tracer))
+        response = app.get("/?foo=bar&x=y")
+
+        assert response.status_int == 200
+        assert response.status == "200 OK"
+
+        spans = tracer.writer.pop_traces()
+        assert len(spans) == 1
+        assert len(spans[0]) == 4
+        request_span = spans[0][0]
+        assert request_span.service == "wsgi"
+        assert request_span.name == "wsgi.request"
+        assert request_span.resource == "GET /"
+        assert request_span.error == 0
+        assert request_span.get_tag("http.method") == "GET"
+        assert request_span.get_tag("http.status_code") == "200"
+        assert request_span.get_tag("http.query.string") == "foo=bar&x=y"
+
+        assert spans[0][1].name == "wsgi.application"
+        assert spans[0][2].name == "wsgi.start_response"
+        assert spans[0][3].name == "wsgi.response"
+
+
+def test_http_request_header_tracing(tracer):
+    config.wsgi.http.trace_headers(["my-header"])
+    app = TestApp(wsgi.DDWSGIMiddleware(application, tracer=tracer))
+    resp = app.get("/", headers={"my-header": "test_value"})
+
+    assert resp.status == "200 OK"
+    assert resp.status_int == 200
+    spans = tracer.writer.pop()
+    assert spans[0].get_tag("http.request.headers.my-header") == "test_value"
+
+
+def test_http_response_header_tracing(tracer):
+    config.wsgi.http.trace_headers(["my-response-header"])
+    app = TestApp(wsgi.DDWSGIMiddleware(application, tracer=tracer))
+    resp = app.get("/", headers={"my-header": "test_value"})
+
+    assert resp.status == "200 OK"
+    assert resp.status_int == 200
+
+    spans = tracer.writer.pop()
+    assert spans[0].get_tag("http.response.headers.my-response-header") == "test_response_value"
+
+
+def test_service_name_can_be_overriden(tracer):
+    with override_config("wsgi", dict(service_name="test-override-service")):
+        app = TestApp(wsgi.DDWSGIMiddleware(application, tracer=tracer))
+        response = app.get("/")
+        assert response.status_code == 200
+
+        spans = tracer.writer.pop_traces()
+        assert len(spans) > 0
+        span = spans[0][0]
+        assert span.service == "test-override-service"
+
+
+def test_chunked_response(tracer):
```

Comment:
Let's copy this out as a snapshot test as well to capture the wsgi.response span that is generated.